backend/regalloc/regalloc_split.ml

```diff
@@ -132,7 +132,55 @@ let make_spill :
     ~id:(State.get_and_incr_instruction_id state)
     ~copy ~from:new_reg ~to_:stack_reg
 
-(* Inserts spills at the end of blocks, before destruction points. *)
+(* Inserts the spills in a block, as early as possible (i.e. immediately after
+   the register is last set), to reduce live ranges. *)
+let rec insert_spills_in_block :
+    State.t ->
+    block_subst:Substitution.t ->
+    stack_subst:Substitution.t ->
+    Cfg.basic_block ->
+    Instruction.t DLL.cell option ->
+    Reg.Set.t ->
+    unit =
+ fun state ~block_subst ~stack_subst block cell live_at_destruction_point ->
+  match Reg.Set.is_empty live_at_destruction_point with
+  | true -> ()
+  | false -> (
+    match cell with
+    | None ->
+      Reg.Set.iter
+        (fun old_reg ->
+          let new_reg = Substitution.apply_reg block_subst old_reg in
+          let copy = block.terminator in
```

Comment:
just to confirm: is this intentionally using the debuginfo of the terminator and not the first instruction in the block? in particular, the spill may be unrelated to this block's terminator, if it was moved up from another block by #1741, right?

Reply:
That's a good point; that code was written
with https://github.com/ocaml-flambda/flambda-backend/pull/1654 in mind, but it was rejected.

Reply:
I think this will be fixed in #1740, not here, right?

Reply:
It is - it did not felt worth it to fix it in both pull requests.